### PR TITLE
Fix workspace delete action bugs

### PR DESCRIPTION
**Pull request by olexii4**
This PR brings several improvements:

Improve delete workspace action:
   added notifications in the case with success and error.
   stopped running workspace before delete.

Improve the logs widget:
  added an error from the WebSocket into the console.
  added error color exactly for errors.

Added additional test coverage for the delete action.

### CODE REVIEW

File: src/components/LogsTab/index.tsx [hunk 1]
```diff
@@ -79,19 +79,18 @@ export class LogsTab extends React.PureComponent<Props, State> {
         this.setState({ isStopped });
       }
 
-      if (workspacesLogs) {
-        const logs = workspacesLogs.get(workspaceId);
-        if (logs && (logs.length !== this.state.logs.length)) {
-          this.setState({ logs });
-        }
+      const logs = workspacesLogs.get(workspaceId);
+      if (this.state.logs !== logs) {
```
Comment (akurinnoy) [inline src/components/LogsTab/index.tsx:87]: I don't think you can compare objects of type `string[] | undefined` in this way.

**Comment by vitaliy-guliy** [comment]
It would be nice to have some description.

File: src/components/Workspace/DeleteAction/index.tsx [hunk 2]
```diff
@@ -88,7 +154,9 @@ class WorkspaceDeleteAction extends React.PureComponent<Props, State> {
 
 }
 
-const mapStateToProps = () => ({});
+const mapStateToProps = (state: AppState) => ({
+  allWorkspaces: selectAllWorkspaces(state),
```
Comment (akurinnoy) [inline src/components/Workspace/DeleteAction/index.tsx]: Why don't you pass a workspace status as component property instead of selecting all workspaces?

File: src/pages/WorkspacesList/index.tsx [hunk 3]
```diff
@@ -10,41 +10,74 @@
...
+export class WorkspacesList extends React.PureComponent<Props, State> {
   private debounce: Debounce;
+  private readonly hideAlert: () => void;
+  private readonly showAlert: (title: string, variant?: AlertVariant) => void;
 
   constructor(props: Props) {
     super(props);
 
+    this.state = {
+      alertVisible: false,
```
Comment (akurinnoy) [inline src/pages/WorkspacesList/index.tsx]: Please, use `Alert` and `AlertGroup` components in accordance to the documentation.

This implementation of workspaces list is able to show various number of alert notifications and attempt to close one of them will obviously hide all notification.

File: src/components/Workspace/DeleteAction/index.tsx (hunk 2 above)
Comment (olexii4) [reply src/components/Workspace/DeleteAction/index.tsx]: Because it is a different behavior depends on status.  I am going to reuse this action now. So, I decided to check the status of the widget.

File: src/store/Workspaces/index.ts [hunk 4]
```diff
@@ -369,12 +377,17 @@ type StatePartial = {
...
+const maxLogLength = 200;
+
 const mapMerge = (originMap: Map<string, string[]>, additionalMap: Map<string, string[]>): Map<string, string[]> => {
   if (!originMap.size) {
     return additionalMap;
   }
   const res = new Map<string, string[]>();
   originMap.forEach((val: string[], key: string) => {
+    if (val.length > maxLogLength) {
+      val.splice(0, maxLogLength);
```
Comment (akurinnoy) [inline src/store/Workspaces/index.ts]: Why do you need to trim long lines? Maybe, splitting a long line in a few short lines would be better? or what about set `overflow: hidden` to the container where logs are displayed?

**Review by akurinnoy — CHANGES_REQUESTED** [review]

File: src/components/Workspace/DeleteAction/index.tsx (hunk 2 above)
Comment (akurinnoy) [reply src/components/Workspace/DeleteAction/index.tsx]: That's what I meant. You don't need to have this component connected at all, just pass the workspace status from parent component.

File: src/pages/WorkspacesList/index.tsx (hunk 3 above)
Comment (olexii4) [reply src/pages/WorkspacesList/index.tsx]: Please, use Alert and Alert Group components in accordance to the documentation. Totally agree Let us discuss

Comment (olexii4) [reply src/pages/WorkspacesList/index.tsx]: My point - it is according to documentation as in other places with show alert

Comment (akurinnoy) [reply src/pages/WorkspacesList/index.tsx]: I meant that on documentation page there are examples of recommended way to use the `AlertGroup` component. Those examples explicitly suggest to manage alert messages in component's state rather than hide the `AlertGroup` in DOM.

Comment (akurinnoy) [reply src/pages/WorkspacesList/index.tsx]: We discussed this offline and reached an agreement to implement in the way the documentation examples provide.

File: src/components/LogsTab/index.tsx (hunk 1 above)
Comment (akurinnoy) [reply src/components/LogsTab/index.tsx:87]: From offline discussion: it's comparing of internal references of two objects. A comment will be added to avoid confusing.

File: src/store/Workspaces/index.ts (hunk 4 above)
Comment (akurinnoy) [reply src/store/Workspaces/index.ts]: We discussed this offline and decided to implement it the way that allows users to download full logs.

File: src/pages/GetStarted/__tests__/GetStarted.spec.tsx [hunk 5]
```diff
@@ -1,89 +0,0 @@
-/*
```
Comment (akurinnoy) [inline src/pages/GetStarted/__tests__/GetStarted.spec.tsx]: Why did you delete this test suit?

File: src/inversify.config.ts [hunk 6]
```diff
@@ -11,16 +11,20 @@
  */
 
 import { Container } from 'inversify';
+// import { NavbarAlerts } from './services/alerts/navbarAlerts';
```
Comment (akurinnoy) [inline src/inversify.config.ts]: please remove commented lines here and below in this config.

Comment (olexii4) [reply src/inversify.config.ts]: I have done

File: src/pages/GetStarted/__tests__/GetStarted.spec.tsx (hunk 5 above)
Comment (olexii4) [reply src/pages/GetStarted/__tests__/GetStarted.spec.tsx]: It was accidentally during rebase. I have returned this code

File: src/components/AppAlertGroup/__tests__/AppAlertGroup.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  });
+
+  it('should show the alert and hide with a close button', () => {
+    const title = 'test 1 message';
+    showAlert(title);
+
+    expect(screen.queryAllByText(title).length).toEqual(1);
+
+    const closeButton = screen.getByRole('button');
+    expect(closeButton).toBeTruthy();
```
Comment (akurinnoy) [inline src/components/AppAlertGroup/__tests__/AppAlertGroup.spec.tsx]: Expecting the `closeButton` to have a truthy value is not necessary here because `getBy*` throws if query matches nothing

```suggestion
```

**Review by akurinnoy — APPROVED** [review]
